### DISCUSSION TITLE
Retrieve abandoned username (2022 version)

### DISCUSSION THREAD

**Discussion by tongsangcong** [community/community#32444] [New to GitHub] [answered]
I wanted to have the abandoned username. I've opened up a ticket for it (Github Support) but haven't gotten an answer. is there any correct way to do this in 2022?

**Comment by garydsong** [comment, marked as answer]
GitHub no longer accepts requests to release dormant usernames .

When you go to contact Support directly about this specifically you will be met with a message:

"It appears that you are looking to contact us regarding releasing a dormant or inactive username. We no longer review individual requests to release dormant or inactive usernames."

There was a point in time where GitHub would release individual usernames depending on inactivity however that time has come and gone. This is a pretty common inquiry so I'm sure GitHub will consider taking action eventually. That being said,  there's no current news about a mass release of dormant accounts but I'm sure one will come in time!

**Reply by ssakib4050** [reply]
I want a username but the username is already taken with no packages no stars or following nothing in the account. 
GitHub should definitely take some action against inactive dormant accounts.

**Reply by StevenRotelli** [reply]
GitHub should consider monetizing this like domain names, if i name is dormant and the user is inactive we should be able to purchase this service. 

I wouldn't recommend paying the owner for the dormant org or user because this would incentivize people to acquire names they don't need, but this would cleanup this userspace IMO.

**Reply by AurelianSpodarec** [reply]
These days you just have to pay a few hundred pounds for a trademark that doens't exist just so you can claim a username for your org on github. Way to go microsoft! LinkedIn has issues too. 

Microsoft is horrible.

**Comment by 54145a** [comment]
I want to add my frustration to this thread, because the current situation is not just inconvenient — it is unfair, and the policy gap makes it worse.

GitHub’s Username Policy explicitly prohibits name squatting. It says account names “may not be reserved or inactively held for future use,” and that violating accounts “may be removed or renamed without notice.” That is clear. But in practice, the only request GitHub will review is a valid trademark complaint. Everything else gets the same answer: “We no longer accept requests to release dormant usernames.”

That means the written rule against squatting is effectively unenforced. And the people who suffer are those who arrived after the early land rush.

It is obvious that many short usernames were claimed in the early days by scripts — automated loops that walked through short strings and treated the namespace as inventory. A huge number of those accounts have no public repos, no commits, no contributions, no visible purpose. They are not using the name. They are just holding it. I am angry about that, and I think that anger is justified. A script should not be able to permanently lock a meaningful identifier away from the person who actually wants to build something with it.

My own case: I chose a short name for a project I am building, and I wanted the matching GitHub organization as its public entry point. The name is held by an account with no public activity and no contact information — no email, no social link. I cannot contact the owner. I cannot buy it, because GitHub prohibits buying or selling usernames and may suspend accounts that try. I cannot file a trademark complaint, because registering a trademark for a name that, as far as public records go, no one else is actively using is not a realistic path for an individual developer. The only official answer left is “pick a different name.”

I did what I could: I secured a matching package scope on npm and a matching organization on Codeberg. But GitHub is still the first place most developers look. Being locked out of a clean identifier because of an early script squat is exactly the kind of thing this policy was supposed to prevent.

I also think the “we cannot determine whether an account is inactive” argument is weaker than it sounds. This is not really about privacy. GitHub does not need to inspect private repositories to know whether an account is still being used. It already has the simplest possible signal: the last login timestamp. An account that has not been logged into for ten years is not being actively used by a human. That is not a subjective judgment. It is an operational fact.

A conservative, automated threshold could work without manual review and without touching private data. For example: no login for 10 years, no active paid subscription, and no recent API activity. Accounts meeting that threshold could have their usernames released automatically. If the concern is false positives, make the threshold extremely conservative. Ten years is already far beyond any reasonable active-use window.

I am not asking GitHub to hand me a specific username. I am asking for one of two things: either enforce the squatting policy that already exists, or implement an automated login-based cleanup so that abandoned short names do not stay locked forever. If GitHub will not review individual dormant accounts, then it should at least stop pretending the squatting rule is meaningful.

Early script squatters win permanently. That is not a fair system for anyone who arrives later.
